restx-jongo-specs-tests/src/main/java/restx/jongo/specs/tests/EmbedMongoFactory.java

```diff
@@ -0,0 +1,77 @@
+package restx.jongo.specs.tests;
+
+import com.mongodb.MongoClientURI;
+import de.flapdoodle.embed.mongo.MongodExecutable;
+import de.flapdoodle.embed.mongo.MongodStarter;
+import de.flapdoodle.embed.mongo.config.IMongodConfig;
+import de.flapdoodle.embed.mongo.config.MongodConfigBuilder;
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.distribution.Version;
+import de.flapdoodle.embed.process.runtime.Network;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.InetAddress;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+class EmbedMongoFactory {
+
+    private static final Map<Version.Main, PoolKey> keys = new ConcurrentHashMap<>();
+    private static final Map<PoolKey, EmbedMongoClientPool> pools = new ConcurrentHashMap<>();
+
+    private static synchronized void newPoolKey(Version.Main version) throws IOException {
+        if (Objects.isNull(keys.get(version))) {
+            InetAddress addr = Network.getLocalHost();
+            int port = Network.getFreeServerPort(addr);
+            String uri = "mongodb://" + addr.getHostName() + ":" + port;
+
+            MongoClientURI mongoClientURI = new MongoClientURI(uri);
+            PoolKey key = new PoolKey(version, mongoClientURI);
+            keys.put(version, key);
+        }
+    }
+
+    PoolKey getPoolKey(Version.Main version) throws IOException {
+        if (Objects.isNull(keys.get(version))) {
+            newPoolKey(version);
+        }
+        return keys.get(version);
+    }
```

Comment:
We have a race condition here: 2 threads can check the isNull in parallel, and then call newPoolKey in parallel. I think you should either synchronize the block or use ConcurrentMap putIfAbsent